clf/nb.go

```diff
@@ -70,21 +73,23 @@ func Create(path *string) Classifier {
 	pipeline := classification.Pipeline
 
 	log.Info("Pipeline:")
-	log.Infof("* RemoveSymbols: \t%v\n", pipeline.RemoveSymbols)
-	log.Infof("* Lower: \t\t%v\n", pipeline.Lower)
-	log.Infof("* Threshold: \t%v\n", pipeline.Threshold)
+	log.Infof("* RemoveSymbols: %v", pipeline.RemoveSymbols)
+	log.Infof("* Lower:         %v", pipeline.Lower)
+	log.Infof("* Threshold:     %v", pipeline.Threshold)
+
+	cls := classification.Classification
 
-	for _, cls := range classification.Classification {
-		for _, txt := range cls.Texts {
-			classifier.Learn(Pipeline(&txt, &pipeline), bayesian.Class(cls.Command))
+	for n := range cls {
+		for i := range cls[n].Texts {
+			classifier.Learn(Pipeline(&cls[n].Texts[i], &pipeline), bayesian.Class(cls[n].Command))
```

Comment:
This was changed because you cannot pass a pointer of a variable that exists only in the for loop. The variables that get created by for loops only exist in the loop. After that they get destroyed. The solution is to pass by index.

Reply:
I come from a Python background, sometimes I have trouble with pointers 😅, thank you for this change and the explanation!